plugins/woocommerce/client/admin/client/marketplace/components/install-flow/install-new-product-modal.tsx

```diff
@@ -46,7 +45,7 @@ function InstallNewProductModal( props: { products: Product[] } ) {
 	const [ showModal, setShowModal ] = useState< boolean >( false );
 	const [ notice, setNotice ] = useState< {
 		message: string;
-		status: Status;
+		status: 'error' | 'info' | 'success' | 'warning' | undefined;
```

Comment:
Could you try extracting these types directly from `React.ComponentProps< typeof Notice >` so they don't need to be manually defined like this? Same with the others (e.g. `type Options`).

Reply:
@mirka Thanks for the review. When I try this, I start getting error, I'm assuming there is no `Notice` type for `React.ComponentProps`

Reply:
What did you try? For example this is working for me:

```suggestion
		status: React.ComponentProps< typeof Notice >[ 'status' ];
```

Reply:
@mirka Thanks! This works. 
Re- your question, I was trying with `React.ComponentProps< typeof Notice >`, it didn't work previously. 

What about Options? `React.ComponentProps< typeof Notice >[ 'options ];` gives me an error. Am I missing something?

Reply:
`options` is not a prop on the [`Notice` component](https://wordpress.github.io/gutenberg/?path=/docs/components-notice--docs), so that is expected.

The original `Options` type is already copied here in the types file, why aren't we reusing that instead of copying it into all the files that use it?

https://github.com/woocommerce/woocommerce/blob/8616ec1398c4d90287a293a42a03fec97f3d2702/plugins/woocommerce/client/admin/client/marketplace/contexts/types.ts#L6